File: day06.py
import sys

input_file = sys.argv[1]
data = open(input_file).read().rstrip().split("\n\n")

print(sum(len(set(group.replace("\n", ''))) for group in data))

print(sum(len(set.intersection(*[set(answers) for answers in group.split("\n")])) for group in data))

File: day06.rb
require 'set'

input_file = ARGV[0]
groups = File.read(input_file).split("\n\n")

puts groups.reduce(0) { | sum, group | sum + group.scan(/[a-z]/).uniq.length }

puts groups.reduce(0) { | sum, group |
    sum + group.split.reduce(Set.new('a'..'z')) { | set, answers | set.intersection(answers.scan(/[a-z]/)) }.length }

